p2p/host/basic/basic_host_test.go

```diff
@@ -359,15 +359,23 @@ func TestHostProtoPreknowledge(t *testing.T) {
 	}
 
 	h2.SetStreamHandler("/super", handler)
-	// Prevent pushing identify information so this test actually _uses_ the super protocol.
-	h1.RemoveStreamHandler(identify.IDPush)
 
 	h1.Start()
 	h2.Start()
 
+	// Prevent pushing identify information so this test actually _uses_ the super protocol.
+	h1.RemoveStreamHandler(identify.IDPush)
```

Comment:
Why after the `Start`? This will cause an Identify Push to be sent.

Reply:
Start in identify is what sets the stream handler. Removing the stream handler before start is no-op since it's not even registered.

Identify won't do anything until these nodes are connected.